app/src/main/java/com/laborofloathing/diceroller/MainActivity.kt

```diff
@@ -27,19 +41,32 @@ class MainActivity : AppCompatActivity() {
             numSides--
             decreaseDiceSize()
         }
-        diceImage = findViewById<ImageView>(R.id.dice_image)
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+
+        Timber.i("onSaveInstanceState called")
+
+        outState.putInt(NUM_SIDES_KEY, numSides)
+        outState.putInt(SIDE_SHOWING, sideShowing)
+        Timber.i("numSides written to bundle")
     }
 
     private fun decreaseDiceSize() {
-        when(numSides) {
+        when (numSides) {
             1 -> decreaseButton.visibility = View.GONE
             else -> decreaseButton.text = getString(R.string.decreasing_button_text, (numSides - 1))
         }
     }
 
     private fun rollDice() {
-        val randomInt = Random().nextInt(numSides) + 1
-        val drawableResource = when(randomInt) {
+        sideShowing = Random().nextInt(numSides) + 1
+        setDiceImage(sideShowing)
+    }
+
+    private fun setDiceImage(sideShowing: Int) {
+        val drawableResource = when (sideShowing) {
```

Comment:
What you did was fine, but, to answer the question you asked in the write-up, you could store `drawableResource` (an integer id specifying the image resource) instead of `sideShowing`.